backend/pkg/modules/internal/fuzzer/tools/secret.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"context"
+
+	uuid "github.com/satori/go.uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	ImagePullSecretNamePrefix = "imagepullsecret"
+	SecretType                = "Opaque"
+	SecretKey                 = "FuzzerAuthData" // nolint:gosec
+)
+
+type ImagePullSecret struct {
+	body      string
+	name      string
+	namespace string
+	key       string
+}
+
+func (s *ImagePullSecret) Name() string {
+	return s.name
+}
+
+func (s *ImagePullSecret) Key() string {
+	return s.key
+}
+
+func (s *ImagePullSecret) Set(value string) {
+	s.body = value
+}
+
+func (s *ImagePullSecret) Save(_ context.Context, client kubernetes.Interface) error {
+	secretDataMap := make(map[string][]byte)
+	secretDataMap[s.key] = []byte(s.body)
+
+	newSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: s.name,
+		},
+		Data: secretDataMap,
+		Type: SecretType,
```

Comment:
`SecretTypeOpaque is the default` no need to set it

Reply:
what to say, here?
Perhaps it is the default , but I'm not aware. I tested various type of secret here to finish by "Opaque". In a personal mind I'm against the fashion to always remove all initializing when it is the default value. For standard type (int, string, etc...) perhaps it is relevant, for a var that have a meaning it's not so obvious and needs to go to the documentation just to read the "default" value.
If you insist I can remove this default value, WDYT?

Reply:
i don't mind you to set it explicitly but IMO better to use the const defined in k8s api `v1.SecretTypeOpaque`

Reply:
ok